src/dev/typescript/ref_output_cache/ref_output_cache.ts

```diff
@@ -130,7 +130,7 @@ export class RefOutputCache {
       }
 
       this.log.debug(`[${relative}] clearing outDir and replacing with cache`);
-      await del(outDir);
+      await del(outDir, { force: true });
```

Comment:
The `outDir` values are all expected to be relative to the repo, in what case would an `outDir` be outside of the repo and require `force`?

Reply:
Okay I think I found the root cause for this. On the current Node.js version on Windows (probably also happening in previous ones), as soon as we require some file/module that is contained inside a symlink it is treated by Windows like it was on a literally different but still equivalent path. In our case as we require `@kbn/optimizer` as part of the setup_node_env the problem starts right here. For example, before requiring the optimizer `__dirname` returns `C:\Users\IEUser\Downloads\kibana` and after the require it while we are inside of it , it starts returning `c:\users\ieuser\downlo~1\kibana`. That will also affect the result of `REPO_ROOT` that we use all over. When running node with `--preserve-symlinks` it stops happening so I think we are on the right path. Aside from that, I was not able to fully bootstrap on windows as the build for `kbn/ui-shared-deps` complains about a node_module that is being loaded from 2 different paths by the same reason listed above and I was in fact able to found old issues about this problem https://github.com/nodejs/node/issues/6624 https://github.com/webpack/webpack/issues/2362. 

For our Bazel work I was already working on https://github.com/elastic/kibana/pull/94046 which might also solve that issue. @peteharverson could you please try that other PR I have open to see if that alternative solution also works for you on Windows?

Reply:
@mistic I gave #94046 a test, but see errors - see https://github.com/elastic/kibana/pull/94046#issuecomment-796606150. Let me know if you want me to try anything else.